source/vir/src/builtins.rs

```diff
@@ -0,0 +1,32 @@
+use crate::ast::{DatatypeTransparency, DatatypeX, GenericBoundX, KrateX, Mode, TypX, Visibility};
+use crate::def::Spanned;
+use air::ast::Span;
+use air::ast_util::ident_binder;
+use std::sync::Arc;
+
+pub fn krate_add_builtins(no_span: &Span, krate: &mut KrateX) {
+    // Add a datatype for 'slice'
+
+    let path = crate::def::slice_type();
+    let visibility = Visibility { owning_module: None, is_private: false };
+    let transparency = DatatypeTransparency::Never;
+
+    // Create a fake variant with a single field of the given type,
+    // though it shouldn't matter, since transparency is Never.
+
+    let typ = Arc::new(TypX::TypParam(crate::def::slice_param()));
+    let field = ident_binder(
+        &Arc::new("DummySliceField".to_string()),
+        &(typ, Mode::Exec, visibility.clone()),
+    );
+    let fields = vec![field];
```

Comment:
It's ok to leave `fields` empty.  This is what we do for external_body datatypes: ```if vattrs.external_body {
        (ident_binder(&variant_name, &Arc::new(vec![])), false)
    } else {```